modules/beta-autopilot-private-cluster/cluster.tf

```diff
@@ -0,0 +1,219 @@
+
+/******************************************
+  Create Container Cluster
+ *****************************************/
+resource "google_container_cluster" "primary" {
+  provider = google-beta
+
+  name            = var.name
+  description     = var.description
+  project         = var.project_id
+  resource_labels = var.cluster_resource_labels
+
+  location          = local.location
+  node_locations    = local.node_locations
+  cluster_ipv4_cidr = var.cluster_ipv4_cidr
+  network           = "projects/${local.network_project_id}/global/networks/${var.network}"
+
+  dynamic "release_channel" {
+    for_each = local.release_channel
+
+    content {
+      channel = release_channel.value.channel
+    }
+  }
+
+  subnetwork = "projects/${local.network_project_id}/regions/${local.region}/subnetworks/${var.subnetwork}"
+
+  default_snat_status {
+    disabled = var.disable_default_snat
+  }
+  min_master_version = var.release_channel != null ? null : local.master_version
+
+  dynamic "cluster_telemetry" {
+    for_each = local.cluster_telemetry_type_is_set ? [1] : []
+    content {
+      type = var.cluster_telemetry_type
+    }
+  }
+  logging_service    = local.cluster_telemetry_type_is_set ? null : var.logging_service
+  monitoring_service = local.cluster_telemetry_type_is_set ? null : var.monitoring_service
+  vertical_pod_autoscaling {
+    enabled = var.enable_vertical_pod_autoscaling
+  }
+  enable_autopilot = var.enable_autopilot
+  dynamic "master_authorized_networks_config" {
+    for_each = local.master_authorized_networks_config
+    content {
+      dynamic "cidr_blocks" {
+        for_each = master_authorized_networks_config.value.cidr_blocks
+        content {
+          cidr_block   = lookup(cidr_blocks.value, "cidr_block", "")
+          display_name = lookup(cidr_blocks.value, "display_name", "")
+        }
+      }
+    }
+  }
+
+  master_auth {
+    username = var.basic_auth_username
+    password = var.basic_auth_password
+
+    client_certificate_config {
+      issue_client_certificate = var.issue_client_certificate
+    }
+  }
+
+  addons_config {
+    http_load_balancing {
+      disabled = ! var.http_load_balancing
+    }
+
+    horizontal_pod_autoscaling {
+      disabled = ! var.horizontal_pod_autoscaling
+    }
+
+    istio_config {
+      disabled = ! var.istio
+      auth     = var.istio_auth
+    }
+
+    dynamic "cloudrun_config" {
+      for_each = local.cluster_cloudrun_config
+
+      content {
+        disabled = cloudrun_config.value.disabled
+      }
+    }
+
+    dns_cache_config {
+      enabled = var.dns_cache
+    }
+
+    dynamic "gce_persistent_disk_csi_driver_config" {
+      for_each = local.cluster_gce_pd_csi_config
+
+      content {
+        enabled = gce_persistent_disk_csi_driver_config.value.enabled
+      }
+    }
+
+    kalm_config {
+      enabled = var.kalm_config
+    }
+
+    config_connector_config {
+      enabled = var.config_connector
+    }
```

Comment:
We may need to validate these. Looks like istio, cloudrun_config,config_connector_config,kalm are not supported.
`dns_cache_config` has to be true.
```
googleapi: Error 400: Addons {"istio"} are not supported for Autopilot clusters., badRequest
```

Reply:
just flipped this chunk off as dns_cache_config by default is set to true